wordpress-mobile/WordPress-iOS
# My Site Dashboard: Fix the dashboard not loading for contributors

Fixes #20593

## Description
Fixes an issue where the dashboard would fail to load properly if the user is a contributor. The root cause was the dashboard endpoint returning errors in place of the expected return types for unsupported cards (stats, pages, activity log). This causes the whole decoding process to fail, even though posts data was returned normally by the endpoint. To fix this, the following was implemented:
* Avoid requesting unsupported cards
* As a safety net, the response decoding now won't fail if one of the cards returns an error, it will simply ignore it. 

## Testing Instructions

1. Log in using an account that's added as a contributor to Site A
2. Switch to site A
3. Navigate to the dashboard
4. Make sure that the dashboard load normally ✅ 
5. Make sure that the stats card is not displayed ✅ 
6. Enable the Pages Card and Activity Card Remote Feature Flags from the debug menu
7. Navigate back to the dashboard
8. Make sure that the dashboard load normally ✅ 
9. Make sure that the pages and activity log cards are not displayed ✅ 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
